src/test/java/core/basesyntax/service/impl/DataConverterImplTest.java

```diff
@@ -0,0 +1,67 @@
+package core.basesyntax.service.impl;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.DataConverter;
+import core.basesyntax.strategy.Operation;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+
+class DataConverterImplTest {
+    private final DataConverter converter = new DataConverterImpl();
+
+    @Test
+    void convert_correctConvert_ok() {
+        List<String> correctList = new ArrayList<>();
+        correctList.add("b,banana,20");
+        correctList.add("s,orange,100");
+        correctList.add("r,apple,100");
+        correctList.add("p,apple,50");
+        List<FruitTransaction> expectedFruitTransactionList = new ArrayList<>();
+        expectedFruitTransactionList.add(new FruitTransaction(Operation.BALANCE, "banana", 20));
+        expectedFruitTransactionList.add(new FruitTransaction(Operation.SUPPLY, "orange", 100));
+        expectedFruitTransactionList.add(new FruitTransaction(Operation.RETURN, "apple", 100));
+        expectedFruitTransactionList.add(new FruitTransaction(Operation.PURCHASE, "apple", 50));
+
+        List<FruitTransaction> actual = converter.convert(correctList);
+
+        for (int i = 0; i < expectedFruitTransactionList.size(); i++) {
+            assertEquals(expectedFruitTransactionList.get(i).getFruit(),
+                    actual.get(i).getFruit());
+            assertEquals(expectedFruitTransactionList.get(i).getOperation(),
+                    actual.get(i).getOperation());
+            assertEquals(expectedFruitTransactionList.get(i).getQuantity(),
+                    actual.get(i).getQuantity());
+        }
```

Comment:
Look for a better way to compare 2 lists. Comparing them one by one in a for loop like this is not a good approach.